File: data-ingestion/src/data_ingestion/app/api/api_v1/models/autocomplete.py
from typing import Optional, List
from pydantic import BaseModel


class QueryModel(BaseModel):
    query: str


class SecurityFunctions(QueryModel):
    pass


class TradeableAssets(BaseModel):
    query_type: Optional[str]
    query_tick: Optional[str]
    
    
class ETFInfoRequest(BaseModel):
    request: dict

class ETFInfoTable(BaseModel):
    ticker: str
    issuer: str
    brand: str
    structure: str
    expense_ratio: str
    etf_home_page: str
    inception: str
    index_tracked: str
    category: str
    asset_class: str
    asset_class_size: str
    asset_class_style: str
    region_general: str
    region_specific: str
    segment: str
    focus: str
    niche: str
    strategy: str
    weighting_scheme: str
    description: str
    alternative_etfs: List[str]
    other_alternative_etfs: List[str]

File: ion-frontend/src/endpoints/schema/autocomplete.ts
import { ASSET_TYPES } from 'common/constant';

export interface ETFInfoRequestProps {
    filter: { [field: string]: string }[];
    sort: { [field: string]: -1 | 1 }[];
}

export interface ETFInfoDTO {
    id: number;
    ticker: string;
    issuer: string;
    brand: string;
    structure: string;
    expense_ratio: string;
    inception: string;
    index_tracked: string;
    category: string;
    asset_class: typeof ASSET_TYPES;
    asset_class_size: string;
    asset_class_style: string;
    region_general: string;
    region_specific: string;
    segment: string;
    focus: string;
    niche: string;
    strategy: string;
    weighting_scheme: string;
    description: string;
}
